markdown.dtx

```diff
@@ -15641,6 +15765,79 @@ function M.writer.new(options)
 % \par
 % \begin{markdown}
 %
+% Define \luamdef{writer->jekyllData} as a function that will transform an
+% input YAML table `d` to the output format. The table is the value of
+% the key `p` in a parent table; if `p` is nil, then the table has no parent.
+% Any string values encountered in the table will be transformed using the
+% function `t`.
+%
+% \end{markdown}
+%  \begin{macrocode}
+  function self.jekyllData(d, t, p)
+    if not self.is_writing then return "" end
+
+    local buf = {}
+
+    local keys = {}
+    for k, _ in pairs(d) do
+      table.insert(keys, k)
+    end
+    table.sort(keys)
+
+    if not p then
+      table.insert(buf, "\\markdownRendererJekyllDataBegin")
+    end
+
+    if #d > 0 then
+        table.insert(buf, "\\markdownRendererJekyllDataSequenceBegin{")
+        table.insert(buf, p or "")
+        table.insert(buf, "}{")
+        table.insert(buf, #keys)
+        table.insert(buf, "}")
+    else
+        table.insert(buf, "\\markdownRendererJekyllDataMappingBegin{")
+        table.insert(buf, p or "")
+        table.insert(buf, "}{")
+        table.insert(buf, #keys)
+        table.insert(buf, "}")
+    end
+
+    for _, k in ipairs(keys) do
+      local v = d[k]
+      local typ = type(v)
+      if typ == "table" then
+        table.insert(
+          buf,
+          self.jekyllData(v, t, k)
+        )
+      elseif typ == "string" or typ == nil then
+        table.insert(buf, "\\markdownRendererJekyllDataString{")
+        table.insert(buf, k)
+        table.insert(buf, "}{")
+        table.insert(buf, t(v or ""))
```

Comment:
For `t` equal to `parse_blocks`, `t(v)` will be fully escaped. However, perhaps we don't want to be too opinionated and we could, instead, provide two variants of the argument: one interpreted as Markdown and fully escaped and one *raw* with as little changes as possible (the `self.uri` escaper is quite gentle) for inclusion to the PDF metadata for example.